### PR TITLE
ATT&CK v9 docker rules

### PR DESCRIPTION
**Pull request by williballenthin**
Here are four rules that cover the ATT&CK v9 additions for container services (i.e. docker and k8s). 

They're not super great because we don't really have any example samples for these techniques - docker/k8s is more commonly used with Linux/MacOS. Still, its not unreasonable that some PE files may interact with docker/k8s, so I've tried to capture the most likely strategies here. 

A bunch of string combinations :-/

I spent some time poking around VT but didn't find any relevant samples to reference as examples.

### CODE REVIEW

File: nursery/build-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      - Defense Evasion::Build Image on Host [T1612]
+    examples:
+    references:
+      - https://docs.docker.com/engine/api/v1.24/
+  features:
+    - or:
+      - string: /^docker(\.exe)? build/
+      - and:
+        - match: send HTTP request
+        - string: /\/v1\.[0-9]{1,2}\/build/
```
Comment (williballenthin) [inline nursery/build-docker-image.yml:17]: 🤮 im sorry

Comment (williballenthin) [reply nursery/build-docker-image.yml:17]: its unclear to me if you can access the API without the version prefix. the docs indicate that you should send it.

File: nursery/build-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+          description: docker API endpoint, e.g., /v1.24/build
```
Comment (mr-tz) [inline nursery/build-docker-image.yml:18]:
```suggestion
          description: docker API endpoint, e.g., /v1.24/build

```

Comment (mr-tz) [reply nursery/build-docker-image.yml:18]: need empty line at EOF, see #291

File: nursery/create-container.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      - string: /^docker(\.exe)? create/
+      - string: /^docker(\.exe)? start/
+      - and:
+        - match: send HTTP request
+        - string: /\/v1\.[0-9]{1,2}\/containers\/create/
+          description: docker API endpoint, e.g., /v1.24/containers/create
+      - and:
+        - match: send HTTP request
+        - string: /\/v1\.[0-9]{1,2}\/containers\/[0-9a-fA-F]+\/start/
+          description: docker API endpoint, e.g., /v1.24/containers/e90e34656806/start
```
Comment (mr-tz) [inline nursery/create-container.yml:23]:
```suggestion
          description: docker API endpoint, e.g., /v1.24/containers/e90e34656806/start

```

File: nursery/list-containers.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    examples:
+    references:
+      - https://docs.docker.com/engine/api/v1.24/
+  features:
+    - or:
+      - string: /^docker(\.exe)? ps/
+      - and:
+        - match: send HTTP request
+        - string: /\/v1\.[0-9]{1,2}\/containers\/json
+          description: docker API endpoint, e.g., /v1.24/containers/json?all=1&before=8dfafdbc3a40&size=1
```
Comment (mr-tz) [inline nursery/list-containers.yml:18]:
```suggestion
          description: docker API endpoint, e.g., /v1.24/containers/json?all=1&before=8dfafdbc3a40&size=1

```

File: nursery/run-in-container.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      - string: /^kubectl(\.exe)? exec/
+      - string: /^kubectl(\.exe)? run/
+      - and:
+        - match: send HTTP request
+        - string: /\/v1\.[0-9]{1,2}\/containers\/[0-9a-fA-F]+\/exec/
+          description: docker API endpoint, e.g., /v1.24/containers/e90e34656806/exec
+      - and:
+        - match: send HTTP request
+        - string: /\/v1\.[0-9]{1,2}\/exec\/[0-9a-fA-F]+\/start/
+          description: docker API endpoint, e.g., /v1.24/exec/e90e34656806/start
```
Comment (mr-tz) [inline nursery/run-in-container.yml:24]:
```suggestion
          description: docker API endpoint, e.g., /v1.24/exec/e90e34656806/start

```

**Review by mr-tz — CHANGES_REQUESTED** [review]
I'm no docker expert and have never seen this in a sample. However, these make sense to me.

Requested changes only on the formatting (the linter should really complain).

**Comment by mr-tz** [comment]
I think some meta data also needs to be reordered, so best to use capafmt.

**Comment by williballenthin** [comment]
reformatted

File: nursery/list-containers.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      - Discovery::Container and Resource Discovery [T1609]
+    references:
+      - https://docs.docker.com/engine/api/v1.24/
+    examples:
+  features:
+    - or:
+      - string: /^docker(\.exe)? ps/
+      - and:
+        - match: send HTTP request
+        - string: /\/v1\.[0-9]{1,2}\/containers\/json
```
Comment (mr-tz) [inline nursery/list-containers.yml:17]: `WARN: rule string quotes incorrect: add double quotes to "/\/v1\.[0-9]{1,2}\/containers\/json"`

Ending `/` is missing.

Comment (williballenthin) [reply nursery/list-containers.yml]: ok linter finally passed! thanks for your patience...

**Review by mr-tz — APPROVED** [review]

**Pull request merged**